sdk/storage/azure-storage-queue/migration_guide.md

```diff
@@ -0,0 +1,577 @@
+# Storage Queue Service SDK Migration Guide from <= 2.x to 12.x
+
+In this section, we list the main changes you need to be aware of when converting your Storage Queue SDK library from version <= 2.X to version 12.X.
+In version 12 we also support asynchronous APIs.
+
+## Converting Core Classes
+<= 2.X synchronous classes have been replaced. Some functionality has been split into separate clients for more logical organization.
+
+| <= 2.X Classes (Clients)  | V12 Clients | NEW Asynchronous clients |
+|---:|---:|---:|
+| QueueService (account-level operations) | QueueServiceClient | QueueServiceClient |
+| QueueService (queue-level operations) | QueueClient   | QueueClient |
```

Comment:
- For the async clients in this table, let's put `aio.QuereServiceClient` etc. to show they live in a different namespace.